src/Plugin/Commerce/ShippingMethod/CanadaPost.php

```diff
@@ -198,13 +166,9 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
   public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
     $values = $form_state->getValue($form['#parents']);
 
-    $this->configuration['api_information']['username'] = $values['api_information']['username'];
-    $this->configuration['api_information']['password'] = $values['api_information']['password'];
-    $this->configuration['api_information']['customer_number'] = $values['api_information']['customer_number'];
-    $this->configuration['api_information']['mode'] = $values['api_information']['mode'];
-    $this->configuration['api_information']['origin_postal_code'] = $values['api_information']['origin_postal_code'];
-    $this->configuration['option_codes'] = $values['option_codes']['codes'];
-    $this->configuration['options']['log'] = $values['options']['log'];
+    $this->configuration['shipping_information']['origin_postal_code'] = $values['shipping_information']['origin_postal_code'];
+    // Remove the empty options codes.
+    $this->configuration['shipping_information']['option_codes'] = array_diff($values['shipping_information']['option_codes'], ['0']);
```

Comment:
@thejacer87 You are changing the plugin's configuration properties, have you considered the implications on existing sites?

1. I believe the log option was not doing anything previously so it's probalby OK.
2. Where the credentials used? Or they would always be fetched from the main config? In the former case sites will break, not sure if we can make that change.
3. Same with options code and postal code, existing sites will break.